File: app/src/main/java/com/cappielloantonio/play/subsonic/models/PodcastEpisode.java
package com.cappielloantonio.play.subsonic.models;

import java.time.LocalDateTime;

public class PodcastEpisode extends Child {
    protected String streamId;
    protected String channelId;
    protected String description;
    protected PodcastStatus status;
    protected LocalDateTime publishDate;

    /**
     * Gets the value of the streamId property.
     *
     * @return possible object is
     * {@link String }
     */
    public String getStreamId() {
        return streamId;
    }

    /**
     * Sets the value of the streamId property.
     *
     * @param value allowed object is
     *              {@link String }
     */
    public void setStreamId(String value) {
        this.streamId = value;
    }

    /**
     * Gets the value of the channelId property.
     *
     * @return possible object is
     * {@link String }
     */
    public String getChannelId() {
        return channelId;
    }

    /**
     * Sets the value of the channelId property.
     *
     * @param value allowed object is
     *              {@link String }
     */
    public void setChannelId(String value) {
        this.channelId = value;
    }

    /**
     * Gets the value of the description property.
     *
     * @return possible object is
     * {@link String }
     */
    public String getDescription() {
        return description;
    }

    /**
     * Sets the value of the description property.
     *
     * @param value allowed object is
     *              {@link String }
     */
    public void setDescription(String value) {
        this.description = value;
    }

    /**
     * Gets the value of the status property.
     *
     * @return possible object is
     * {@link PodcastStatus }
     */
    public PodcastStatus getStatus() {
        return status;
    }

    /**
     * Sets the value of the status property.
     *
     * @param value allowed object is
     *              {@link PodcastStatus }
     */
    public void setStatus(PodcastStatus value) {
        this.status = value;
    }

    /**
     * Gets the value of the publishDate property.
     *
     * @return possible object is
     * {@link String }
     */
    public LocalDateTime getPublishDate() {
        return publishDate;
    }

    /**
     * Sets the value of the publishDate property.
     *
     * @param value allowed object is
     *              {@link String }
     */
    public void setPublishDate(LocalDateTime value) {
        this.publishDate = value;
    }
}

File: app/src/main/java/com/cappielloantonio/play/subsonic/models/PodcastEpisode.kt
package com.cappielloantonio.play.subsonic.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
class PodcastEpisode : Parcelable {
    var id: String? = null

    @SerializedName("parent")
    var parentId: String? = null

    @SerializedName("isDir")
    var isDir = false
    var title: String? = null
    var album: String? = null
    var artist: String? = null
    var track: Int? = null
    var year: Int? = null
    var genre: String? = null

    @SerializedName("coverArt")
    var coverArtId: String? = null
    var size: Long? = null
    var contentType: String? = null
    var suffix: String? = null
    var transcodedContentType: String? = null
    var transcodedSuffix: String? = null
    var duration: Int? = null
    var bitRate: Int? = null
    var path: String? = null

    @SerializedName("isVideo")
    var video: Boolean? = null
    var userRating: Int? = null
    var averageRating: Double? = null
    var playCount: Long? = null
    var discNumber: Int? = null
    var created: Date? = null
    var starred: Date? = null
    var albumId: String? = null
    var artistId: String? = null
    var type: String? = null
    var bookmarkPosition: Long? = null
    var originalWidth: Int? = null
    var originalHeight: Int? = null
    var streamId: String? = null
    var channelId: String? = null
    var description: String? = null
    var status: String? = null
    var publishDate: Date? = null
}